d2renderers/d2svg/d2svg.go

```diff
@@ -1830,15 +1831,15 @@ func Render(diagram *d2target.Diagram, opts *RenderOpts) ([]byte, error) {
 	}
 
 	var dimensions string
-	if setDimensions {
+	if setDimensions && !opts.NoFit {
 		dimensions = fmt.Sprintf(` width="%d" height="%d"`, w, h)
 	}
```

Comment:
I experimented with applying this change in a diagram and learned that it does not do what I expected. Please see the note here: https://github.com/terrastruct/d2/issues/1080#issuecomment-1482221132